clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2261,82 +2322,62 @@ public DescribeConsumerGroupsResult describeConsumerGroups(final Collection<Stri
 
             final String groupId = entry.getKey();
 
-            final long startFindCoordinatorMs = time.milliseconds();
-            final long deadline = calcDeadlineMs(startFindCoordinatorMs, options.timeoutMs());
+            final long nowDescribeConsumerGroups = time.milliseconds();
+            final long deadline = calcDeadlineMs(nowDescribeConsumerGroups, options.timeoutMs());
+
+            Node coordinator = coordinators.get(groupId);
+            int nodeId = coordinator == null ? -1 : coordinator.id(); // leave null-handling to the next NodeProvider
 
-            runnable.call(new Call("findCoordinator", deadline, new LeastLoadedNodeProvider()) {
+            runnable.call(new Call("describeConsumerGroups", deadline, new ConstantNodeIdProvider(nodeId)) {
```

Comment:
@hachikuji Correct me if I am wrong. Coordinator-finding should be always finished before doing group-related tasks, no matter async or sync interface we use, so it means we have to wait for it in any cases. What we should do is to ensure the blocking time is not unlimited.

Reply:
@hachikuji is correct.  We can't do blocking operations in the admin client service thread.  We certainly can't do blocking operations that wait for the service thread itself.  This will deadlock.

I think it's a good idea to have a coordinator node provider, but we need to build out a little more infrastructure to make it possible.  I have a change which should help with that, at https://github.com/apache/kafka/pull/4295

Reply:
@hachikuji Do you think it's okay to convert this async operation into a sync one, similar with this above:
```
TopicPartition tp = new TopicPartition("__consumer_offsets", Math.abs(groupID.hashCode() % 50));
return metadata.fetch().leaderFor(tp);
```

Reply:
It's an interesting thought, but users may override the number of partitions for `__consumer_offsets`, so I don't think it will work. More generally, we are trying to avoid dependence in the clients on the `__consumer_offsets` topic since it ties the behavior of the client to what is more properly an implementation detail.